services/order.py

```diff
@@ -0,0 +1,40 @@
+from typing import Optional
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Ticket, Order
+
+
+@transaction.atomic
+def create_order(
+    tickets: list[dict[str, int]],
+    username: str,
+    date: Optional[str] = None
+) -> None:
+    order = Order.objects.create(
+        user=get_user_model()
+        .objects.get(username=username)
+    )
+
+    if date:
+        order.created_at = date
+        order.save()
+
+    for ticket in tickets:
+        Ticket.objects.create(
+            order=order,
+            row=ticket["row"],
+            seat=ticket["seat"],
+            movie_session_id=ticket["movie_session"]
+        )
+
+
+def get_orders(username: Optional[str] = None) -> QuerySet:
+    orders = Order.objects.all()
+    if username:
+        user = get_user_model().objects.get(username=username)

```

Comment:
you can do a lookup on `orders` queryset that you assigned above. Filter by `user__username`,